### ISSUE TITLE
Strange behaviour with Popup's tooltip background color

### ISSUE THREAD

**Issue by Nightfallen** [ocornut/imgui#4729]
**Version/Branch of Dear ImGui:**

Version: 1.86 WIP (18510)
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_dx11.cpp + imgui_impl_win32.cpp
Compiler: MSVC
Operating System: Windows 10

**My Issue/Question:**

After clicking on color preview when using 'ImGui::ColorEdit4' or 'ImGui::ColorEdit3' on popup windows the tooltip's background color is changing to 'ImGuiCol_ModalWindowDimBg'.
But when i open ColorEdit its background color isn't changing to 'ImGuiCol_ModalWindowDimBg'
Maybe when tooltip's rect is out of bound of window and the new viewport is created (as imgui metrics/debugger says) it isn't properly configured or missing some flags?

Also when using several windows with popups ImGui::PushStyleCol(ImGuiCol_ModalWindowDimBg, color) doesn't work.

**Screenshots/Video**

![strange](https://user-images.githubusercontent.com/93279017/141957539-e97d8b9f-5ed2-4f5d-bd2b-ed0ccbb81cbe.gif)

**Standalone, minimal, complete and verifiable example:** 
```cpp
void SomeImguiWindow(bool* is_open)
{
	static bool just_once = true;
	if (just_once)
	{
		ImGui::SetNextWindowSize(ImVec2(1000, 600));
		just_once = false;
	}
	//ImGui::PushStyleColor(ImGuiCol_ModalWindowDimBg, { 0.f, 0.f, 0.f, 0.8f });
	auto& io = ImGui::GetIO();
	auto& style = ImGui::GetStyle();
	ImGui::Begin("##window_no_name1", is_open);
	static bool popup = false;

	if (ImGui::Button("Open popup"))
		popup = true;

	if (!ImGui::IsPopupOpen("Popup") && popup)
	{
		ImGui::OpenPopup("Popup");
		ImGui::SetNextWindowSize({ 500.f, 300.f });
	}

	style.Colors[ImGuiCol_ModalWindowDimBg] = { 0.f, 0.f, 0.f, 0.8f };
	if (ImGui::BeginPopupModal("Popup", is_open, ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoResize))
	{
		static float col1[3] = { 1.0f, 0.0f, 0.2f };
		static float col2[4] = { 0.4f, 0.7f, 0.0f, 0.5f };
		ImGui::ColorEdit3("Color 1", col1);
		ImGui::ColorEdit4("Color 2", col2);

		ImGui::EndPopup();
	}
	ImGui::End();
	//ImGui::PopStyleColor();
}
```

**Comment by ocornut** [comment]
Hello,

If I paste your code I don't seem to see the issue. Does it actually reproduce for you with this exact code?
Are you using multi-viewports and do you see a difference whether multi-viewports are enabled or disabled?
Can you paste the details found in the About box?

(
PS: unrelated to you specific issue, but here are some code suggestion:
```cpp
static bool just_once = true;
if (just_once)
{
	ImGui::SetNextWindowSize(ImVec2(1000, 600));
	just_once = false;
}
```
Can become
```
ImGui::SetNextWindowSize(ImVec2(1000, 600), ImGuiCond_Once);
```

And
```
if (ImGui::Button("Open popup"))
	popup = true;
if (!ImGui::IsPopupOpen("Popup") && popup)
{
	ImGui::OpenPopup("Popup");
	ImGui::SetNextWindowSize({ 500.f, 300.f });
}
```
Can become
```
if (ImGui::Button("Open popup"))
	ImGui::OpenPopup("Popup");
ImGui::SetNextWindowSize({ 500.f, 300.f }, ImGuiCond_Appearing);
```
)

**Comment by Nightfallen** [comment]
Thanks, didn't know ImGuiCond exists.
Yes, I'm using multi-viewports. Without multi-viewports everything works as intended.
Also I'm using a few hacks to hide main winapi window from screen and its taskbar icon with this code, but i don't think it has effect on issue.
```cpp
int wnd_wh = 100;
hwnd_ = CreateWindowEx(WS_EX_TOOLWINDOW, wc_.lpszClassName, wndName, WS_POPUP | WS_SYSMENU, 100, 100, wnd_wh, wnd_wh, NULL, NULL, wc_.hInstance, NULL);

// Make main window's area fully transparent
SetWindowLong(hwnd_, GWL_EXSTYLE, (int)GetWindowLong(hwnd_, GWL_EXSTYLE) | WS_EX_LAYERED | WS_EX_TRANSPARENT);
SetLayeredWindowAttributes(hwnd_, RGB(0, 0, 0), 0, LWA_ALPHA);
```
I know there is 'io.ConfigViewportsNoDefaultParent' flag and 'ImGui_ImplWin32_CreateWindow' but i don't know how to properly use them to avoid creating winapi window myself.


Info from About box:

```
Dear ImGui 1.86 WIP (18510)
--------------------------------

sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _WIN64
define: _MSC_VER=1931
define: _MSVC_LANG=202004
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------

io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx11
io.ConfigFlags: 0x00000401
 NavEnableKeyboard
 ViewportsEnable
io.ConfigViewportsNoAutoMerge
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigWindowsMoveFromTitleBarOnly
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------

io.Fonts: 2 fonts, Flags: 0x00000000, TexSize: 512,512
io.DisplaySize: 50.00,50.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------

style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 5.00,2.00
style.FrameRounding: 3.00
style.FrameBorderSize: 1.00
style.ItemSpacing: 6.00,6.00
style.ItemInnerSpacing: 6.00,6.00
```

**Comment by ocornut** [comment]
Bug confirmed with `io.ConfigViewportsNoAutoMerge = true`, will investigate.

**Comment by ocornut** [comment]
So, the bug technically has nothing to do with the modal or multi-viewports, they just make the bug visible.

The bug is that Tooltip windows are not consistently moved to the front of the array corresponding to display order (it's only during render that they are made part of a separate layer) and modal rendering uses that data to decide what to dim.
In `master` branch looking at metrics we can see how Tooltips are not at the top-most spot in the `g.Windows[]` array.

I don't think there is a good reason for things working this way. It's just lazy logic that came in to existence in 2014 (6267905a). I'm going to rewire things to make them consistent with the `g.Windows[]` and perhaps we can even remove the layering thing done in Render()..

**Comment by ocornut** [comment]
Pushed a fix a3667f462a01b5dfc05e8032a2006762c426d4fd
